include/hgwosca.h

```diff
@@ -18,7 +18,9 @@ double * gwo_hgwosca(obj_func_t obj_func,
    Initialise population of `wolf_count` wolves, each with `dim` dimensions, where
    each dimension is bound by `min_positions` and `max_positions`.
  */
-double *gwo_init_population(size_t wolf_count, size_t dim, const double min_position, const double  max_position);
+void gwo_init_population(double *population,
+                         size_t wolf_count, size_t dim,
+                         const double min_position, const double  max_position);
```

Comment:
You probably want to remove the const from the function declaration but do as you wish. Also, a copy happens anyway so max_position cannot be changed outside of this function. I think in this case the const only states that the copy _inside_ this function will not change inside the body, would u agree?

Reply:
Yeah exactly. So it would not compile (or at least throw a warning) if the value is changed within the function. It would not affect the algo in any way if it did, because it is a copy, but it is a safeguard in the sense that it should not happen because you have no reason for the value to change in the function.

Reply:
Ok cool. So in case you want to have this safeguard in the implementation you would stick to const double arg, however, as we know by now, this const expression is ignored in the function declaration and does not play a role for the interface of the function so one would probably drop it there.